frontend/packages/console-shared/src/hooks/usePinnedResources.ts

```diff
@@ -13,18 +13,55 @@ type PinnedResourcesType = {
 
 const PINNED_RESOURCES_CONFIG_MAP_KEY = 'console.pinnedResources';
 
+const getCustomizedPins = (id: string) => {
+  if (window.SERVER_FLAGS.perspectives) {
+    const perspectives: Perspective[] = JSON.parse(window.SERVER_FLAGS.perspectives);
+    const perspective = perspectives.find((p: Perspective) => p.id === id && p?.pinnedResources);
```

Comment:
Multiple perspectives with the same id aren't allowed, so you can just look up by id and then return pinnedResources (as you already do)

```suggestion
    const perspective = perspectives.find((p: Perspective) => p.id === id);
```